inc/namespace.php

```diff
@@ -924,60 +926,94 @@ function elasticpress_mapping( array $mapping, ?string $index = null ) : array {
-		if ( ! $user_dictionary_package_id && $is_network_japanese ) {
-			$user_dictionary_package_id = Packages\get_package_id( 'uploaded-user-dictionary', true );
-		}
-
-		// Check for a package ID and add it to the kuromoji tokenizer.
-		if ( $user_dictionary_package_id ) {
-			$mapping['settings']['analysis']['tokenizer']['kuromoji']['user_dictionary'] = $user_dictionary_package_id;
-		}
-	}
-
 	// Add a default search analyzer if any custom stopwords or synonyms are provided.
 	//
 	// Synonyms and stopwords are quick enough to be applied at search time and avoid
 	// increasing the index size unnecessarily.
 	$is_network_language = get_site_option( 'WPLANG', 'en_US' ) === get_option( 'WPLANG', 'en_US' );
 	$synonyms = [];
 	$stopwords = [];
+	$kuromoji_dictionary = null;
+	$inline_index_settings = should_inline_settings();
 
-	foreach ( [ 'synonyms', 'stopwords' ] as $type ) {
+	foreach ( [ 'synonyms', 'stopwords', 'user-dictionary' ] as $type ) {
 		foreach ( [ 'uploaded', 'manual' ] as $sub_type ) {
 			// Get package file path.
-			$package_id = Packages\get_package_id( "{$sub_type}-{$type}" );
+			$package_slug = "{$sub_type}-{$type}";
+			$package_id = Packages\get_package_id( $package_slug );
+			$is_network_package = false;
 			// Check for network default.
 			if ( ! $package_id && $is_network_language ) {
-				$package_id = Packages\get_package_id( "{$sub_type}-{$type}", true );
+				$package_id = Packages\get_package_id( $package_slug, true );
+				$is_network_package = true;
 			}
 
 			// Check for a package ID.
 			if ( ! $package_id ) {
 				continue;
 			}
 
+			if ( $inline_index_settings ) {
+				$package_contents = get_package_contents( $package_slug, $is_network_package );
+				if ( ! $package_contents ) {
+					trigger_error( sprintf( 'Could not fetch %s package contents of "%s".', $type, $package_id ) );
+					continue;
+				}
+				$package_contents = preg_split( '[\r\n]', $package_contents );
+			} else {
+				$package_contents = null;
+			}
+
 			switch ( $type ) {
 				case 'synonyms':
-					$synonyms[ "{$sub_type}_{$type}_filter" ] = [
-						'type' => 'synonym_graph',
-						'synonyms_path' => $package_id,
-					];
+					if ( $inline_index_settings ) {
+						$synonyms[ "{$sub_type}_{$type}_filter" ] = [
+							'type' => 'synonym_graph',
+							'synonyms' => $package_contents,
+						];
+					} else {
+						$synonyms[ "{$sub_type}_{$type}_filter" ] = [
+							'type' => 'synonym_graph',
+							'synonyms_path' => $package_id,
+						];
+					}
 					break;
 				case 'stopwords':
-					$stopwords[ "{$sub_type}_{$type}_filter" ] = [
-						'type' => 'stop',
-						'ignore_case' => true,
-						'stopwords_path' => $package_id,
-					];
+					if ( $inline_index_settings ) {
+						$stopwords[ "{$sub_type}_{$type}_filter" ] = [
+							'type' => 'stop',
+							'ignore_case' => true,
+							'stopwords' => implode( ',', $package_contents ),
```

Comment:
Any reason not to pass this as an array as well? Should be supported iirc.

Reply:
True, the docs just referenced a string so i went with that, but looking again it does indeed support arrays as well.

Reply:
```suggestion
							'stopwords' => $package_contents,
```